src/thor/map_matcher.cc

```diff
@@ -183,51 +188,64 @@ MapMatcher::FormPath(meili::MapMatcher* matcher,
   // Iterate through the matched path. Form PathInfo - populate elapsed time
   // Return an empty path (or throw exception) if path is not connected.
   Cost elapsed;
-  std::vector<PathInfo> path;
+  std::deque<std::pair<std::vector<PathInfo>, std::vector<const meili::EdgeSegment*>>> paths;
   GraphId prior_edge, prior_node;
   EdgeLabel pred;
-  auto prev_segment_target = std::numeric_limits<float>::max();
+  const meili::EdgeSegment* prev_segment = nullptr;
+  const GraphTile* tile = nullptr;
+  const DirectedEdge* directededge = nullptr;
+  const NodeInfo* nodeinfo = nullptr;
 
   // Build the path
   size_t interpolated_index = 0;
+  size_t num_segments = edge_segments.size();
+  Cost accumulated_elapsed;
   for (const auto& edge_segment : edge_segments) {
-    // Skip edges that are the same as the prior edge if they are not disconnected,
-    // Note that, when the prior edge and current edge has the same edgeid and the current
-    // source is smaller than prior target, it indicates discontinuity. see follow:
-    //
-    //                   current_source
-    //    prior_source           |    prior_target
-    //          |                |         |
-    //  X---------------------------------------------X
-    //                      Edge
-    if (edge_segment.edgeid == prior_edge && prev_segment_target <= edge_segment.source) {
-      continue;
-    }
 
     // Get the directed edge
     GraphId edge_id = edge_segment.edgeid;
     matcher->graphreader().GetGraphTile(edge_id, tile);
     directededge = tile->directededge(edge_id);
 
     // Check if connected to prior edge
-    bool disconnected = false;
-    if (prior_edge.Is_Valid() &&
-        !matcher->graphreader().AreEdgesConnectedForward(prior_edge, edge_id)) {
-      disconnected_edges.emplace_back(prior_edge, edge_id);
-      disconnected = true;
+    bool disconnected =
+        prev_segment != nullptr && prev_segment->edgeid.Is_Valid() && prev_segment->discontinuity;
+
+    bool break_point =
+        edge_segment.first_match_idx >= 0 &&
+        (options.shape(edge_segment.first_match_idx).type() == Location::kBreak ||
```

Comment:
dont actually need to use options.shape because the match result now has this info